tv_shows/index.js

```diff
@@ -0,0 +1,158 @@
+let mockReviews = [
+    {
+        text: "This is the best show I've ever seen. Everything is connected!",
+        rating: 5
+    },
+    {
+        text: "This is so dark, I love it!",
+        rating: 5
+    },
+]
+let starsSelected = 0;
+
+function createReviewItem(review) {
+    const reviewItemElement = document.createElement('div');
+    reviewItemElement.classList = ['review-item'];
+
+    const reviewTextItemElement = document.createElement('div');
+    reviewTextItemElement.classList = ['review-text-item'];
+    reviewTextItemElement.textContent = review.text;
+    reviewItemElement.appendChild(reviewTextItemElement);
+
+    const reviewFoterItemElement = document.createElement('div');
+    reviewFoterItemElement.classList = ['review-footer-item'];
+    reviewItemElement.appendChild(reviewFoterItemElement);
+
+    const reviewRatingItemElement = document.createElement('div');
+    reviewRatingItemElement.classList = ['review-rating-item'];
+    const rating = review.rating;
+    for (let index = 0; index < rating; index++) {
+        const filledStar = createFilledStarElement();
+        reviewRatingItemElement.appendChild(filledStar);
+    };
+    for (let index = 0; index < 5 - rating; index++) {
+        const emptyStar = createEmptyStarElement();
+        reviewRatingItemElement.appendChild(emptyStar);
+    };
+    const reviewRatingValue = document.createElement('div');
+    reviewRatingValue.classList = ['review-rating-value'];
+    reviewRatingValue.textContent = rating.toString().concat('/5');
+    reviewRatingItemElement.appendChild(reviewRatingValue);
+    reviewFoterItemElement.appendChild(reviewRatingItemElement);
+
+    const deleteReviewButtonElement = document.createElement('button');
+    deleteReviewButtonElement.classList = ['delete-review-button'];
+    deleteReviewButtonElement.textContent = 'Delete';
+    deleteReviewButtonElement.onclick = () => {
+        mockReviews = mockReviews.filter((r) => {
+            return r !== review;
+        });
+        renderReviewList();
```

Comment:
Kod dodavanja i brisanja nam se uvijek sve rerendera, a to želimo malo optimizirat